packages/xsnap/src/netstring.js

```diff
@@ -72,6 +72,7 @@ export async function* reader(input, name = '<unknown>', capacity = 1024) {
       `Unexpected dangling message at offset ${offset} of ${name}`,
     );
   }
+  return new Uint8Array(0);
```

Comment:
For a JS generator, the `yield X` results in `{ done: false, value: X }` and `return X` results in `{ done: true, value: Y }`.  The value `Y` is not a signal, it's just an empty value with the same type as `X` so that Typescript doesn't get confused.